Sources/SwiftCliCore/StitchLibrary.swift

```diff
@@ -50,3 +48,33 @@ func isStitchValid(stitch: String) -> Bool {
     }
     return false
 }
+
+func isStitchAndStitchCountValid(stitch: String, rowNumber: Int, stitchIndex: Int) -> Result<String, InputError> {
```

Comment:
This method is quite long and a little hard to follow. Could you take a stab at breaking it up?